web-ng/src/app/components/map/map.component.ts

```diff
@@ -81,13 +83,39 @@ export class MapComponent implements AfterViewInit, OnDestroy {
   }
 
   ngAfterViewInit() {
+    const dummyProject: Project = new Project(
+      '',
+      StringMap({ en: 'title' }),
+      StringMap({ en: 'description' }),
+      Map({}),
+      Map({})
+    );
     this.subscription.add(
       combineLatest([
-        this.activeProject$,
-        this.features$,
-      ]).subscribe(([project, features]) =>
-        this.onProjectAndFeaturesUpdate(project, features)
-      )
+        this.activeProject$.pipe(startWith(dummyProject)),
+        this.features$.pipe(startWith(List<Feature>())),
+      ])
+        .pipe(pairwise())
+        .pipe(
+          map(([[oldProject, oldFeatures], [newProject, newFeatures]]) => {
```

Comment:
Rather than look at the previous value received in the stream, reactive UI elements should  compare their current UI state with the state from the incoming event. The idea is that events in the stream are idempotent and independent, which affords us some guarantees that make the code foolproof and debuggable. Would it be possible to compare against the features in the map instead of the previous value in the stream?

Reply:
Good point, now comparing based on the UI state(this.markers for location feature & this.map.data for geo json feature) and values from the stream.